### ISSUE TITLE
Process a process definition deletion

### ISSUE THREAD

**Issue by remcowesterhoud** [camunda/zeebe#9769]
# Description
We need to create a processor to process the deletion of a process definition.

- Extend the processor for the resource deletion command
    - Find the resource in the state
    - Write a `ResourceDeletion DELETED` event
    - Write a `Process DELETED` event
    - Distribute the command

Out of scope:
- Decision definition deletion

Blocked by: #9766 #9767
